src/main/features/offer/form/models/offer.ts

```diff
@@ -17,24 +17,27 @@ export class Offer implements Serializable<Offer> {
   @IsDefined({ message: ValidationErrors.OFFER_REQUIRED })
   @IsNotBlank({ message: ValidationErrors.OFFER_REQUIRED })
   @MaxLength(99000, { message: ValidationErrors.OFFER_TEXT_TOO_LONG })
-  offerText?: string
+  offerText: string
 
   @ValidateNested()
   @IsDefined({ message: ValidationErrors.DATE_REQUIRED })
   @IsValidLocalDate({ message: ValidationErrors.DATE_NOT_VALID })
   @IsFutureDate({ message: ValidationErrors.FUTURE_DATE })
-  completionDate?: LocalDate
+  completionDate: LocalDate
 
-  constructor (offerText?: string, completionDate?: LocalDate) {
-    this.offerText = offerText
-    this.completionDate = completionDate
+  static create (offerText: string, completionDate: LocalDate) {
```

Comment:
What is the purpose of this? Why did you introduce it over regular constructor? If there is no significant advantage of this I would suggest reverting back to constructor as this is the pattern / expected way of instantiating classes.